pyproject.toml

```diff
@@ -74,3 +74,205 @@ multi_line_output = 3
 include_trailing_comma = true
 combine_as_imports = true
 skip = "src/twisted/logger/__init__.py,src/twisted/internet/reactor.py"
+
+
+[tool.tox]
+legacy_tox_ini = """
+; tox configuration file for running tests.
+;
+; The main trial based environments are defined based on a set of factors. Each
+; factor will trigger a different behavior. Multiple factors can be set for the
+; same environment:
+;
+; * alldeps - install all possible dependencies.
+; * nodeps - avoid installing any dependencies apart from testing tools.
+; * withcov - run the tests wrapped using the coverage.
+; * nocov - run the tests directly, without using the coverage wrapper.
+; * wheel - build the wheel distribution
+; * posix - prepare the tests to be run in a Linux/Unix/macOS environment.
+; * macos - prepare the tests to be run in a macOS 10.11+ environment (superset of posix)
+; * windows - prepare the tests to be executed under Windows.
+;
+; See README.rst for example tox commands.
+;
+; There are also various non-default environments used by the continuous
+; integration system: the `codecov-push` and `coveralls-push` push the coverage
+; results to codecov.io and coveralls.io, respectively. They should be called
+; after running both some number of `-withcov` environments and also
+; `coverage-prepare`.
+;
+; For compatibility with the current infrastructure, `codecov-publish`
+; combines `coverage-prepare` and `codecov-push` into a single step.
+;
+; A non-default `txchecker-travis` environment is used to run twistedchecker
+; on travis in --diff mode
+;
+[tox]
+minversion=3.20.1
+requires=
+    virtualenv>=20.0.35
+    tox-wheel>=0.5.0
+skip_missing_interpreters=True
+envlist=lint, mypy, black,
+    apidocs, narrativedocs, newsfragment,
+    manifest-checker, twine,
+    py38-alldeps-nocov
+
+[default]
+; Files and directories that contain Python source for linting.
+sources = setup.py src/ docs/conch/examples docs/mail/examples docs/names/examples docs/pair/examples docs/web/examples docs/words/examples
+; These examples fail lint, see https://twistedmatrix.com/trac/ticket/9952
+;   docs/core/examples
+
+[testenv]
+wheel = True
+wheel_pep517 = True
+;; dependencies managed by extras in setup.cfg
+extras =
+    ; The "nodeps" build still depends on PyHamcrest.
+    nodeps: test
+
+    alldeps: all_non_platform
+
+    windows: windows_platform
+
+    alldeps-macos: osx_platform
+
+    serial: serial
+
+    ; Documentation needs Twisted install to get the version.
+    narrativedocs: dev
+
+;; dependencies that are not specified as extras
+deps =
+    {withcov}: coverage
+
+    {coverage-prepare,codecov-publish}: coverage
+
+    {codecov-push,codecov-publish}: codecov
+
+    coveralls-push: coveralls
+    coveralls-push: PyYAML
+
+    wheel: wheel
+
+    ; Code quality checkers
+    manifest-checker: check-manifest>=0.42
+
+    twine: twine
+
+    lint: pyflakes
+
+    black: black==20.8b1
+
+    wheels: cibuildwheel==1.6.1
```

Comment:
cibuildwheel is here :)

I think is best to keep the tox.ini in the old location and have a separate pr to clean it